src/Compilers/CSharp/Portable/Syntax/InternalSyntax/SyntaxList`1.cs

```diff
@@ -9,10 +9,12 @@
 using Microsoft.CodeAnalysis.Text;
 using Roslyn.Utilities;
 using System.Diagnostics;
+using Microsoft.CodeAnalysis.Syntax.InternalSyntax;
 
 namespace Microsoft.CodeAnalysis.CSharp.Syntax.InternalSyntax
 {
-    internal partial struct SyntaxList<TNode> : IEquatable<SyntaxList<TNode>> where TNode : CSharpSyntaxNode
+    internal partial struct SyntaxList<TNode> : IEquatable<SyntaxList<TNode>>, ISyntaxList<TNode> 
+        where TNode : CSharpSyntaxNode
```

Comment:
@cston My end goal is that SyntaxList would be in core.  However, that's going to take several ronuds of refactorings to get there.  The reason for this is that SyntaxList is pretty tightly entertwined with the different languages.  For example, the lists subclass the language specific nodes.  They also have smarts baked into them in terms of how and what weak references are used.

I ahve these (followup) PRs that continue the refactorings:
https://github.com/dotnet/roslyn/pull/13122
https://github.com/dotnet/roslyn/pull/13124
https://github.com/dotnet/roslyn/pull/13128
https://github.com/dotnet/roslyn/pull/13129
https://github.com/dotnet/roslyn/pull/13144

As i get these in, it will make it easier and easier to move things like SyntaxList and SyntaxListBuilder over to core.